### PR TITLE
Fix manifest_version as a section bug.

### PR DESCRIPTION
**Pull request by Integralist**
## Problem

There was a report by someone at Fastly earlier in the week regarding an issue with the `0.25.0` CLI release. The issue was caused by an existing compute project they had which used a fastly.toml without a `manifest_version` set.

Ultimately the end result was that the `manifest_version` field was being stored in the fastly.toml as a 'section', e.g. `[manifest_version]` which completely changes the interpretation of the toml file and would cause the toml parser to error when encountering it.

Removing `[manifest_version]` before running `compute build` fixed the issue because the fixes in `0.26.0` meant we would persist the correct `manifest_version` value back to the manifest file, but this is still confusing behaviour for any users who might have stumble into this.

## Solution

Before we attempt to unmarshal the toml data we first need to identify if the manifest contains a `manifest_version` as a 'section'.

Once we've identified if it is set as a section (e.g. `[manifest_version]`), we can then proceed to remove it from the file.

## Notes

I had initially hoped to have used the toml parsing library to support the deletion of the `[manifest_version]` section (e.g. delete the section and then use the existing Write() method to overwrite the manifest file) but unfortunately because I can't guarantee the location of the manifest_version field within the manifest file it could mean we accidentally end up deleting a larger chunk of the configuration.

For example...

```
authors = ["integralist@fastly.com"]
description = "..."
language = "rust"
[manifest_version]
name = "..."
service_id = "..."
```

Using the toml parser to delete the `[manifest_version]` would cause `name` and `service_id` to also be deleted as they are mistakenly interpreted as being _within_ that same section block.

This means I have to take a different approach, which is to read the manifest line by line and to remove the `[manifest_version]` that way.

### CODE REVIEW

**Review by triblondon — APPROVED** [review]

**Review by phamann — APPROVED** [review]
I'm still personally on the fence if we should even ship this, without other systems in place to ensure we will delete the code in a timely fashion. However, I do understand that this is a genuine user issue in the wild and thus happy to review.

TL;DR looks like a clean implementation to me 👍🏻 have left a couple of minor style nits but nothing blocking.

File: pkg/compute/manifest/manifest.go [hunk 1]
```diff
@@ -229,6 +257,59 @@ func (f *File) Read(fpath string) error {
...
+	}
+
+	if _, ok := tree.GetArray("manifest_version").(*toml.Tree); ok {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// stripManifestSection reads the manifest line-by-line storing the lines that
```
Comment (doramatadora) [inline pkg/compute/manifest/manifest.go:275]:
```suggestion
// It would've been better if we could have relied on the toml library to delete
```

File: pkg/compute/manifest/manifest.go [hunk 2]
```diff
@@ -229,6 +254,60 @@ func (f *File) Read(fpath string) error {
...
+	for scanner.Scan() {
+		if scanner.Text() != "[manifest_version]" {
+			_, err := buf.Write(scanner.Bytes())
+			if err != nil {
+				return buf, err
+			}
+			_, err = buf.WriteString("\n")
+			if err != nil {
+				return buf, err
+			}
```
[the remark this answers is not in the document]
Comment (doramatadora) [reply pkg/compute/manifest/manifest.go:298]: I have a not-a-go-developer question if you go with @phamann's suggestion: why is the second argument of append `[]byte("\n")` (not `[]byte("\n")...`)?

**Review by doramatadora — APPROVED** [review]
Looks good to my non-SME eyes — spotted a typo!
